### ISSUE TITLE
[Feature Request] Variables + Template

### ISSUE THREAD

**Issue by FelipeRearden** [getmatterapp/obsidian-matter#4]
Hello @hunterclarke and @tianskylan  !!!!!!

First of all, thank you very much for doing this integration with Obsidian. I have been using Matter for the past 6 months I am pretty happy to finally integrate my two favorite Apps. Thank you very much  !!!!

And .... Welcome to our community :)

## MY FR 

One of the most important things in Obsidian is to organize your notes to further search. When you have many notes, you have to define certain patterns inside each note to have an easier way to find what you need. 

MY FR is simple to explain: would be amazing to have variables `{{variable}}` and a template to customize the notes that are imported/created in Obsidian by the Matter sync. 

### Variables

Based on my experience with Matter I think we can have theses variables:

{{title}} -> the tittle of the article 
{{url}} -> http://www.google.com
{{author}} -> name of the autor 
{{site}} -> name of the article
{{publish_date}} -> YYY-MM-DD
{{highlights}}
{{tags:separator}} -> 
- Obsidian uses tags for many things. Would be amazing to have in the note tags that was inserted in Matter. 
- If a tag have more than one name (Joe Biden) the whitespace needs to be replaced by a `_` `Joe_Biden`
- For YAML the syntax in Obsidian is 
```
[tag1,tag2]
```

{{#tags:separator}} -> 
- Obsidian uses tags for many things. Would be amazing to have in the note tags that was insert in Matter. 
- If a tag have more than one name (Joe Biden) the whitespace needs to be replaced by a `_` `Joe_Biden`
- For non-YAML the syntax in Obsidian is
```
#tag1 #tag2
```

I'm gonna use this new article as an example:

![416A8E21-E2A8-4867-900B-76FF81B377C0](https://user-images.githubusercontent.com/85180696/145565468-a04008b6-03cb-4b56-b2a7-9be22300665d.jpeg)

![1A6B1C6D-2C31-4FC6-9FB3-41B461E157B2](https://user-images.githubusercontent.com/85180696/145565474-919bd350-f457-4375-be6a-0005ab7c8786.jpeg)

{{title}} -> What I told students of Princeton
{{url}} -> http://www.google.com
{{author}} -> Abigail Shrier
{{site}} -> The Truth Fairy
{{publish_date}} -> 2021-12-08
{{highlights}}
{{tags:,}} -> Columbia_University,Habits,Productivity
{{#tags:\s}} -> #Columbia_University #Habits #Productivity

## Template 

This is a template UI from another plugin that I think is the best to show my template FR:

![0](https://user-images.githubusercontent.com/85180696/145566170-7f4825d1-0d2b-41dd-a579-7dae417bfb39.png)

Following my example. I would have this template for my notes:

```
---
tags: [{{tags:,}}]
---

<kbd id=tit>{{title}}</kbd><strong id=lit>[{{author}} - {{site}}](https://www.google.com)</strong>

## Highlights
{{highlights}}

## Outline

```
that generates this note in Obsidian like this:

![E366B2C8-34D3-4288-9400-DE6D1A1FCB63](https://user-images.githubusercontent.com/85180696/145566233-9614c068-5764-40cd-ae5b-dad65001839d.jpeg)

---

I hope you like my suggestion ! Let me know if you need more information !!!

Thanks for reading this!!!!!

Have a great day!

**Comment by marcjulianschwarz** [comment]
I would like to second this feature request.

Matter is the first „read later“-app that I enjoy using and the plugin for Obsidian is the cherry on the cake.
As @FelipeRearden already wrote down all the details I can only say that I would love to see this feature coming to the plugin.
Thank you very much for the effort you have put into this.

**Comment by hunterclarke** [comment]
Thanks for the thorough issue report @FelipeRearden! Templates with all available metadata (including tags) are certainly something on my own personal wish list as well. We will keep you all posted.

**Comment by FelipeRearden** [comment]
Thank you very much @hunterclarke !!!!!!!

I'm happy to know that this FR is in your wish list :) 

Would be a great update for all Obsidian users. 

Let me know if you need more information. I'm always here :)


Thanks again and have a great day!

**Comment by dolphinpoint** [comment]
Chiming to add my support for the feature request. The ability to customize metadata (and filename) would be fantastic. I'd like to be able to set the following settings.

```
filename: {{YYYYMMDDHHmmSS}}

---
title: {{title}}
aliases:
tags: 
url: {{url}}
---


{{highlights}}
```

**Comment by creativefibro** [comment]
I came here to ask for Templates too.  Also, the Matter sync folder will not allow you to use a nested folder as other input plugins do, which means that it sticks out in my folder structure. 

Matter, is my favourite choice as a reader app out of all of them, but for the lack of playing nicely with the rest of my  Obsidian set-up.

**Comment by stefanopagliari** [comment]
I also would like to second this request. Being able to create a template would be incredibly helpful as it would allow to integrate the plugin with the rest of my Obsidian set-up.

**Comment by pwflint** [comment]
Upvoting per the comments above.

**Comment by hunterclarke** [comment]
Hey folks, I wanted to reach out and provide an update. I think we're almost to the point where I can start working on variables & templating for the plugin. I know it's been a long time coming. We've been working hard behind the scenes to integrate with more services and land on an approach that would generalize and be maintainable over time. @FelipeRearden, your initial message here is a huge first step towards that solution.

The last of the variables for our static template have been implemented [here](https://github.com/getmatterapp/obsidian-matter/pull/9). That will be released later this week. Feedback welcome!

Once that has been merged and released, I'll begin work on the code for template support. If anyone has any additional input please leave a comment here. Thanks for your patience on this!

---

@creativefibro the issue with your folders should have been fixed with version 1.0.6. Please open a new issue if you're still having problems.

**Comment by pwflint** [comment]
Updated this morning and ran a few test articles and everything works smoothly. Thanks for the hard work, really looking forward to support for YAML.

PWF

**Comment by philoserf** [comment]
👍🏻 for user control via a template

an example, as provided by a kindle plugin, is located here
https://github.com/hadynz/obsidian-kindle-plugin/blob/2baa56653e2e4c6dfa545cc7a1cd300b9ff81e01/src/rendering/index.ts

**Comment by hunterclarke** [comment]
Big news, the PR for templates is open: https://github.com/getmatterapp/obsidian-matter/pull/25. I'll be testing it over the next few days. Please feel free to add any feedback.

**Comment by hunterclarke** [comment]
Templates has officially been released with version 1.1.0. You can go ahead and upgrade. There is also another issue for tracking additional fields you would like to see in the templates [here](https://github.com/getmatterapp/obsidian-matter/issues/26). Let me know if you have any feedback.

Excited to close this one out!
